### PR TITLE
Distributed survey

### PR DESCRIPTION
**Pull request by panyx0718**

### CODE REVIEW

File: doc/fluid/design/dist_train/distributed_traing_review.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+# Data Parallelism
+Data Parallelism runs the same model on multiple devices, each taking in a partition of the input batch. It’s more commonly used for a few reasons. It generally applies to common SGD mini-batch training. Compared with model parallelism, which requires users to carefully partition their model and tune for good performance, data parallelism usually involves no more than calling an extra API and speed up is more predictable.
+
+# Asynchronous Training
+In asynchronous training, it usually involves a set of trainers and a set of parameter servers. The parameter servers collectively hold a single copy of sharedsharded parameters. While the trainers each holds a unique copy of model and trains the model independently. Each trainer pulls parameters from parameter servers and sends gradients to the parameter servers independently. Similarly the parameter servers applies the gradients to parameters as soon as the gradients are received and sends parameters whenever they are requested.
```
Comment (jacquesqiao) [inline doc/fluid/design/dist_train/distributed_traing_review.md]: sharedsharded  => shared

File: doc/fluid/design/dist_train/distributed_traing_review.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Many production models, such as [3], are trained with distributed asynchronous solutions due to its scalability and effectiveness in practice. However, asynchronous training has its limitations. Usually, it’s not as stable as synchronous training. A warm-up phase is sometimes needed. Learning rate is usually smaller compared with synchronous training and decay is also often needed. Normally, asynchronous training doesn’t scale beyond 100 trainers. In other words, when putting more trainers beyond that, the model cannot converge faster.
+
+# Synchronous Training
+Unlike asynchronous training, synchronous training requires step barriers. Parameter servers needs to wait for gradients from all trainers before they are applied to parameters and trainers will always pull the latest parameters.
+
+An obvious advantage of synchronous training is that the behavior is more clearly defined. Usually, it more stable than asynchronous training. Learning rate can be set larger and for some vision tasks, the final accuracy can be slightly higher. (In my practical experience, for some models, it can actually be worse).
```
Comment (jacquesqiao) [inline doc/fluid/design/dist_train/distributed_traing_review.md]: it more => it's more

Comment (panyx0718) [reply doc/fluid/design/dist_train/distributed_traing_review.md]: Done

File: doc/fluid/design/dist_train/distributed_traing_review.md (hunk 1 above)
Comment (panyx0718) [reply doc/fluid/design/dist_train/distributed_traing_review.md]: Done

**Pull request merged**